src/common/commonutils/SshUtils.c

```diff
@@ -292,6 +292,7 @@ static int CheckOnlyApprovedMacAlgorithmsAreUsed(const char* macs, char** reason
         return status;
     }
 
+    sshMacs = DuplicateStringToLowercase(g_sshMacs);
```

Comment:
Add another 'else if' case after line 293, check there for successful allocation for sshMacs and if failed, log and return ENOMEM. Like this:

else if (NULL == (sshMacs = DuplicateStringToLowercase(g_sshMacs)))
    {
        OsConfigLogError(log, "CheckOnlyApprovedMacAlgorithmsAreUsed: failed to duplicate string to lowercase");
        return ENOMEM;
    }